homeassistant/components/starlink/sensor.py

```diff
@@ -40,21 +40,17 @@ async def async_setup_entry(
 ) -> None:
     """Set up all sensors for this entry."""
     async_add_entities(
-        StarlinkSensorEntity(config_entry.runtime_data, description)
+        description.entity_class(config_entry.runtime_data, description)
         for description in SENSORS
     )
 
-    async_add_entities(
-        StarlinkRestoreSensor(config_entry.runtime_data, description)
-        for description in RESTORE_SENSORS
-    )
-
 
 @dataclass(frozen=True, kw_only=True)
 class StarlinkSensorEntityDescription(SensorEntityDescription):
     """Describes a Starlink sensor entity."""
 
     value_fn: Callable[[StarlinkData], datetime | StateType]
+    entity_class: Callable
```

Comment:
entity_class changes seem like an unrelated change, can we do this in a separate PR?